Code/Tools/AssetProcessor/native/utilities/PlatformConfiguration.cpp

```diff
@@ -1863,7 +1874,57 @@ namespace AssetProcessor
 
     void PlatformConfiguration::AddGemScanFolders(const AZStd::vector<AzFramework::GemInfo>& gemInfoList)
     {
-        int gemOrder = g_gemStartingOrder;
+        // Determine if a Gem is relative to the project. Cross-check gem paths against visited project gem paths.
+        // If the gem is project-relative, make adjustments to it's priority order based on registry settings:
+        // /Amazon/AssetProcessor/Settings/GemScanFolderStartingPriorityOrder
+        // /Amazon/AssetProcessor/Settings/ProjectRelativeGemsScanFolderPriority
+        // See <o3de-root>/Registry/AssetProcessorPlatformConfig.setreg for more information.
+
+        AZ::s64 gemStartingOrder = 100;
+        AZStd::string projectGemPrioritySetting{};
+        AZStd::vector<AZ::IO::Path> projectGemPaths{};
+        int pathCount = 0;
+
+        const int projectScanOrder = GetProjectScanFolderOrder();
+
+        auto const settingsRegistry = AZ::SettingsRegistry::Get();
+        if (settingsRegistry != nullptr)
+        {
+            settingsRegistry->Get(gemStartingOrder,
+                AZ::SettingsRegistryInterface::FixedValueString(AssetProcessorSettingsKey) + "/GemScanFolderStartingPriorityOrder");
+
+            settingsRegistry->Get(projectGemPrioritySetting,
+                AZ::SettingsRegistryInterface::FixedValueString(AssetProcessorSettingsKey) + "/ProjectRelativeGemsScanFolderPriority");
+            AZStd::to_lower(projectGemPrioritySetting.begin(), projectGemPrioritySetting.end());
+
+            auto projectGemCallback = [&projectGemPaths]([[maybe_unused]] AZStd::string_view manifestObjectKey,
+                [[maybe_unused]] AZStd::string_view manifestObjectName, AZStd::string_view manifestRootPath)
+            {
+                projectGemPaths.push_back(AZ::IO::Path{ manifestRootPath });
+            };
+            AZ::SettingsRegistryMergeUtils::VisitProjectGems(*settingsRegistry, projectGemCallback);
```

Comment:
Instead of gathering a list of project gem paths from the SettingsRegistry, we can go with the cheaper option of just querying the project path.
```
AZ::IO::FixedMaxPath projectPath = AZ::Utils::GetProjectPath();
```

Reply:
Simplified to this and used `IsRelativeTo` to check.